src/hydration/water-quality.ts

```diff
@@ -0,0 +1,106 @@
+import { Item } from 'net.md_5.bungee.api.chat.hover.content';
+import { Bukkit } from 'org.bukkit';
+import { Biome, Block, BlockFace } from 'org.bukkit.block';
+import { PlayerInteractEvent } from 'org.bukkit.event.player';
+import { ItemStack } from 'org.bukkit.inventory';
+import { PotionMeta } from 'org.bukkit.inventory.meta';
+import { PotionData, PotionType } from 'org.bukkit.potion';
+
+export type WaterQuality = 'CLEAR' | 'DIRTY' | 'SALTY' | 'NORMAL';
+
+const WATER_POTION_DATAS = new Map<WaterQuality, PotionData>([
+  ['CLEAR', new PotionData(PotionType.MUNDANE)],
+  ['DIRTY', new PotionData(PotionType.THICK)],
+  ['SALTY', new PotionData(PotionType.AWKWARD)],
+  ['NORMAL', new PotionData(PotionType.WATER)],
+]);
+const POTION_TYPE_QUALITIES = new Map<PotionType, WaterQuality>([
+  [PotionType.MUNDANE, 'CLEAR'],
+  [PotionType.THICK, 'DIRTY'],
+  [PotionType.AWKWARD, 'SALTY'],
+  [PotionType.WATER, 'NORMAL'],
+]);
+const DEFAULT = new PotionData(PotionType.WATER);
+
+// prettier-ignore
+const BIOME_QUALITIES = new Map<Biome, WaterQuality>([
+  [Biome.OCEAN,                 'SALTY'],
+  [Biome.DEEP_OCEAN,            'SALTY'],
+  [Biome.COLD_OCEAN,            'SALTY'],
+  [Biome.DEEP_COLD_OCEAN,       'SALTY'],
+  [Biome.FROZEN_OCEAN,          'SALTY'],
+  [Biome.DEEP_FROZEN_OCEAN,     'SALTY'],
+  [Biome.LUKEWARM_OCEAN,        'SALTY'],
+  [Biome.DEEP_LUKEWARM_OCEAN,   'SALTY'],
+  [Biome.WARM_OCEAN,            'SALTY'],
+  [Biome.DEEP_WARM_OCEAN,       'SALTY'],
+
+  [Biome.SNOWY_MOUNTAINS,       'CLEAR'],
+  [Biome.MOUNTAINS,             'CLEAR'],
+]);
+
+const CLEAR_WATER_HEIGHT = 110;
```

Comment:
This is *minimum* height for clear water in biomes that have it, not a height that automatically makes water clear in all biomes. Please leave a comment indicating so, I was confused for a while.